crates/client/analytics/src/lib.rs

```diff
@@ -0,0 +1,226 @@
+use opentelemetry::metrics::{Counter, Gauge, Histogram, Meter};
+use opentelemetry::trace::TracerProvider;
+use opentelemetry::{global, KeyValue};
+use opentelemetry_appender_tracing::layer::OpenTelemetryTracingBridge;
+use opentelemetry_otlp::{ExportConfig, WithExportConfig};
+use opentelemetry_sdk::logs::LoggerProvider;
+use opentelemetry_sdk::metrics::reader::{DefaultAggregationSelector, DefaultTemporalitySelector};
+use opentelemetry_sdk::metrics::{PeriodicReader, SdkMeterProvider};
+use opentelemetry_sdk::trace::{BatchConfigBuilder, Config, Tracer};
+use opentelemetry_sdk::{runtime, Resource};
+use std::fmt::Display;
+use std::str::FromStr as _;
+use std::time::Duration;
+use tracing::Level;
+use tracing_opentelemetry::OpenTelemetryLayer;
+use tracing_subscriber::layer::SubscriberExt as _;
+use tracing_subscriber::util::SubscriberInitExt as _;
+use tracing_subscriber::EnvFilter;
+use url::Url;
+
+pub struct Analytics {
+    meter_provider: Option<SdkMeterProvider>,
+    service_name: String,
+    log_level: Level,
+    collection_endpoint: Option<Url>,
+}
+
+impl Analytics {
+    pub fn new(service_name: String, log_level: String, collection_endpoint: Option<Url>) -> anyhow::Result<Self> {
+        let log_level = Level::from_str(&log_level).unwrap_or(Level::INFO);
+        Ok(Self { meter_provider: None, service_name, log_level, collection_endpoint })
```

Comment:
```suggestion
    pub fn new(service_name: String, log_level: tracing::Level, collection_endpoint: Option<Url>) -> anyhow::Result<Self> {
        Ok(Self { meter_provider: None, service_name, log_level, collection_endpoint })
```

This seems less error-prone than relying on a string here. You can then use `tracing::Level::ERROR` and related.

Reply:
Valid, moreover I have reworked the `AnalyticsParams` to take in `analytics_log_level` as Level itself.
see [here](https://github.com/madara-alliance/madara/pull/348/commits/476d41186fb42842d63f5b98a62e1e4377f553e4).